### PR TITLE
Add 'follow sprite' command for title sequence

### PR DESCRIPTION
**Pull request by Broxzier**
This PR adds a new command to the title sequence player that can set the viewport to follow a given sprite. The related windows have been updated so that the new command can be added to a sequence easily.

I've made a short video to demonstrate how this can be used: http://www.broxzier.com/files/ShareX/2017-07-17_23-45-04.mp4

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Nice addition! In the video, there appears to be some corruption in drawing the logo as the title screen is following the sprite. I'm assuming that's what 2213a6a08d0a476d808e6276e9861b7b07c5164d is supposed to tackle?

**Comment by marijnvdwerf** [comment]
Would it be possible to have a human-readable sprite description?

File: src/openrct2/title/TitleScreen.cpp [hunk 1]
```diff
@@ -217,6 +217,9 @@ extern "C"
         openrct2_write_full_version_info(ch, sizeof(buffer) - (ch - buffer));
         gfx_draw_string(dpi, buffer, COLOUR_BLACK, x + 5, y + 5 - 13);
 
+        // Invalidate screen area
+        gfx_set_dirty_blocks(x, y, x + 500, y + 30);
```
Comment (janisozaur) [inline src/openrct2/title/TitleScreen.cpp]: `x + 500, y + 30`? That's a bit arbitrary, care to explain why these values exactly?

File: src/openrct2/windows/title_command_editor.c [hunk 2]
```diff
@@ -36,18 +36,20 @@ typedef struct TITLE_COMMAND_ORDER {
...
+    { TITLE_SCRIPT_SPEED,       STR_TITLE_EDITOR_COMMAND_TYPE_SPEED,    STR_TITLE_EDITOR_ARGUMENT_SPEED },
+    { TITLE_SCRIPT_FOLLOW,      STR_TITLE_EDITOR_COMMAND_TYPE_FOLLOW,   STR_NONE },
+    { TITLE_SCRIPT_WAIT,        STR_TITLE_EDITOR_COMMAND_TYPE_WAIT,     STR_TITLE_EDITOR_ARGUMENT_WAIT_SECONDS },
+    { TITLE_SCRIPT_RESTART,     STR_TITLE_EDITOR_RESTART,               STR_NONE },
+    { TITLE_SCRIPT_END,         STR_TITLE_EDITOR_END,                   STR_NONE },
 };
 
-#define NUM_COMMANDS 8
+// Basically countof(window_title_command_editor_orders)
+#define NUM_COMMANDS (sizeof(window_title_command_editor_orders) / sizeof(window_title_command_editor_orders[0]))
```
Comment (janisozaur) [inline src/openrct2/windows/title_command_editor.c]: there's a helper macro `countof`, please use that

**Comment by janisozaur** [comment]
I like @marijnvdwerf 's suggestion too. I think it would be easy to mis-click a sprite and not know about it. Some confirmation either by naming the sprite or flashing it (you could toggle ghost flag, right?) would help alleviate that.

For the former, there should be a generic function that for each sprite would return its name in textual form. I'm actually a bit surprised there isn't one yet.

File: src/openrct2/title/TitleScreen.cpp (hunk 1 above)
Comment (Broxzier) [inline src/openrct2/title/TitleScreen.cpp]: `500` is just a large number to cover the width of the string. AppVeyor builds usually have a long version name for example. If there's a simply way to obtain the exact length, please do tell. `30` is simply the line height, to cover both the version and platform labels.

**Comment by Broxzier** [comment]
@AaronVanGeffen Yes, 2213a6a08d0a476d808e6276e9861b7b07c5164d fixes the drawing issues in the video.
@marijnvdwerf Do you mean user feedback about the selected sprite, or clearer strings? Any suggestions?

**Comment by marijnvdwerf** [comment]
Yeah, talking about clearer strings that `133427`. Thinking of guest names, and "{{ ride.name }} - Train {{ n }}" as seen in the tooltips when you hover over a vehicle.

**Comment by Broxzier** [comment]
8fc980c0783fcf3f0236dd22ad49e97aa08102fd is WIP, just wanted to rebase before continuing work for this PR.

**Comment by Gymnasiast** [comment]
Superseded by #6606.

**Pull request closed without merging**